dbms/src/Flash/Pipeline/Schedule/ThreadPool/TaskThreadPool.cpp

```diff
@@ -74,18 +74,35 @@ void TaskThreadPool<Impl>::doLoop(size_t thread_no)
     LOG_INFO(thread_logger, "start loop");
 
     TaskPtr task;
-    while (likely(task_queue->take(task)))
+    ExecTaskStatus status_after_exec;
+    while (true)
     {
-        metrics.decPendingTask();
-        handleTask(task);
+        try
+        {
+            if unlikely (!task_queue->take(task))
```

Comment:
The function of taskQueue should not throw errors

Reply:
All methods called in TaskThreadPool should not throw errors

Reply:
During the execution of Pipeline Tasks, if resource group is deleted, we have the following options:
1. report error(resource group not found)
    1. throw exception in TaskQueue::take()/submit()/updatestatistics9)
    2. set task a flag, so when exec task, report error
2. let tasks of this query finish without consuming any RU.

Current implement choose 1.1

Reply:
take()/updateStatistics()/submit() will handle exception inside.